R/settings_utils.R

```diff
@@ -68,24 +68,26 @@ find_config <- function(filename) {
     dirname(filename)
   }
 
-  ## check for a file in the current directory
-  linter_config <- file.path(path, linter_file)
-  if (isTRUE(file.exists(linter_config))) {
-    return(linter_config)
-  }
-
-  ## next check for a file higher directories
-  linter_config <- find_config2(path)
-  if (isTRUE(file.exists(linter_config))) {
-    return(linter_config)
-  }
-
-  ## next check for a file in the user directory
-  # cf: rstudio@bc9b6a5 SessionRSConnect.R#L32
-  home_dir <- Sys.getenv("HOME", unset = "~")
-  linter_config <- file.path(home_dir, linter_file)
-  if (isTRUE(file.exists(linter_config))) {
-    return(linter_config)
+  # List possible .lintr file locations
+  file_locations <- c(
+    # Check for a file in the current directory
+    file.path(path, linter_file),
+    # Next check for a file in the .github/linters directory
+    file.path(path, ".github/linters", linter_file),
+    # Next check for a file in higher directories
+    find_config2(path),
+    # Next check for a file in the user directory
+    # cf: rstudio@bc9b6a5 SessionRSConnect.R#L32
+    file.path(Sys.getenv("HOME", unset = "~"), linter_file),
+    # Next check for a global config file
+    file.path(R_user_dir("lintr", which = "config"), "config")
```

Comment:
Maybe use `lintr_file` as the filename here as well?

Also, from the earlier discussion I though that XDG_CONFIG_HOME might also be a useful search location, right?

Reply:
> Maybe use lintr_file as the filename here as well?

To clarify: you think the file path should be e.g. `~/.config/R/lintr/lintr_file` instead of `~/.config/R/lintr/config`? I’m happy with either but the latter is more common (e.g. Git has `~/.config/git/config`, etc).

> Also, from the earlier discussion I though that XDG_CONFIG_HOME might also be a useful search location, right?

`tool::R_user_dir` uses `XDG_CONFIG_HOME`, if set.

Reply:
I was thinking it should be `getOption("lintr.linter_file")` like with all other candidate paths.

Reply:
Hm, I disagree with this. Firstly, I don’t think there’s a benefit to making this configurable (the path *is* already configurable on multiple levels, including via `XDG_CONFIG_HOME` and `R_USER_CONFIG_DIR`).

And secondly, the filename given by `getOption("lintr.linter_file")` makes the file invisible on most platforms (leading dot by default). This is appropriate in the home directory or a project directory (…ish, to be honest). It is *not* appropriate or necessary in a dedicated config directory. Users (me included) might want to use an invisible filename for their projects (e.g. `.lintr`) but a regular file for the global configuration.

What do you think?

Reply:
Okay, works for me.